bin/upversion

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash -e
+#
+# Usage:
+#   bin/upversion NEW_VERSION
+# 
+# Changes the version where it is hardcoded in UDMI files, performing checks
+# that the new version is available, and that the files can be safely updated.
+# Modifications:
+#   - schema/*.json (all files)
+#   - etc/categories.json 
+#   - tests/**/*.json and validator/traces/**/*.out - 
+#     all files must be added to etc/upversion.txt only files preceeded
+#     with a 'y' are updated. Only works with "pretty"/non-minified JSON
+#   - documentation in-line exammples
+#   - specific java files (Pubber.java, ConfigUtil.java, LocalDevice.java)
+
+UPVERSION_LIST=etc/upversion.txt
+TESTS_REGEX="^(\s{0,4}\"version\"\s*:\s*)([0-9.\"]*)(,?)"
+SCHEMA_VERSION_IDENTIFIER=\$udmi_version # must start with $
+
+# Updates existing $udmi_version in a JSON file. Called by 
```

Comment:
NB: Typically avoid "called by" comments because all they do is get out of date and become inaccurate. It's easy enough to search for "update_existing_schema_version" so it's not hard to find where it's used...